distributed/client.py

```diff
@@ -1515,12 +1515,11 @@ async def _close(self, fast=False):
                 _set_global_client(None)
 
             if (
-                not fast
-                and handle_report_task is not None
+                handle_report_task is not None
                 and handle_report_task is not current_task
             ):
                 with suppress(TimeoutError, asyncio.CancelledError):
-                    await asyncio.wait_for(handle_report_task, 2)
+                    await asyncio.wait_for(handle_report_task, 0 if fast else 2)
```

Comment:
currently thinking about how to write a test for this

Reply:
I think the most straightforward one is to throw a spanner in the scheduler so that this line never returns:

https://github.com/graingert/distributed/blob/31cee2016591dc8bfc1e994400c9c347c9b8a365/distributed/client.py#L1371